core/trie/proof_test.go

```diff
@@ -252,6 +252,55 @@ func build4KeyTrie(t *testing.T) *trie.Trie {
 	return tempTrie
 }
 
+func noDuplicates(proofNodes []trie.ProofNode) bool {
+	seen := make(map[string]bool)
+	for _, pNode := range proofNodes {
```

Comment:
I think we can set the len here 	seen := make(map[string]bool, len(proofNodes))
Also, just curious, why map[string]bool instead of map[*felt.Felt]bool? Saves you calling the String() method and is a tiny bit more readable.

Reply:
We changed it to felt.Felt in code but I forgot to change this in tests